src/main/scala/org/broadinstitute/monster/encode/extraction/EncodeExtractions.scala

```diff
@@ -0,0 +1,217 @@
+
+    /**
+      * Construct a future which will either complete with the stringified
+      * payload resulting from querying a url, or fail.
+      *
+      * @param client the HTTP client to use in the query
+      * @param params ???
+      */
+    private def get(client: OkHttpClient, params: List[String]): Future[String] = {
+      val url = s"https://www.encodeproject.org/search/?${params.mkString(sep = "&")}"
+      val promise = Promise[String]()
+
+      val request = new Request.Builder()
+        .url(url)
+        .get
+        .build
+
+      client
+        .newCall(request)
+        .enqueue(new Callback {
+          override def onFailure(request: Request, e: IOException): Unit = {
+            promise.failure(e)
+            ()
+          }
+
+          override def onResponse(response: Response): Unit = {
+            if (response.isSuccessful) {
+              promise.success(response.body.string)
+            } else if (response.code() == 404) {
+              promise.success("""{ "@graph": [] }""")
+            } else {
+              promise.failure(
+                new RuntimeException(s"ENCODE lookup failed: $response")
+              )
+            }
+            ()
+          }
+        })
+
+      promise.future
+    }
+
+    override protected def newClient(): OkHttpClient = client
+  }
+
+  /**
+    * Pipeline stage which maps batches of search parameters into JSON entities
+    * from ENCODE matching those parameters.
+    *
+    * @param encodeEntity the type of ENCODE entity the stage should query
+    */
+  def getEntities(
+    encodeEntity: EncodeEntity
+  ): SCollection[List[(String, String)]] => SCollection[JsonObject] =
+    _.transform(s"Download ${encodeEntity.entryName} Entities") {
+      _.applyKvTransform(ParDo.of(new EncodeLookup(encodeEntity))).flatMap { kv =>
+        kv.getValue.fold(
+          throw _,
+          value => {
+            val decoded = for {
+              // TODO change to upack and msg instead of circe and json, dear future monster
+              json <- io.circe.parser.parse(value)
+              cursor = json.hcursor
+              objects <- cursor.downField("@graph").as[Vector[JsonObject]]
+            } yield {
+              objects
+            }
+            decoded.fold(throw _, identity)
+          }
+        )
+      }
+    }
+
+  /**
+    * Pipeline stage which extracts IDs from downloaded JSON entities for
+    * use in subsequent queries.
+    *
+    * @param entryName display name for the type of entity whose IDs will
+    *                  be extracted in this stage
+    * @param referenceField field in the input JSONs containing the IDs
+    *                       to extract
+    * @param manyReferences whether or not `referenceField` is an array
+    */
+  def getIds(
+    entryName: String,
+    referenceField: String,
+    manyReferences: Boolean
+  ): SCollection[JsonObject] => SCollection[String] =
+    _.transform(s"Get $entryName IDs") { collection =>
```

Comment:
I would ditch this `transform` call in favor of more specific calls at the top of the pipeline.